Add client wrapper for MongoDB Atlas support

zenmoto opened the pull request:
**Description:** 

Adds pagination, error handling, will be extended to manage metric
transformation.

zenmoto committed:
client wrapper for MongoDB Atlas support

Adds pagination, error handling, will be extended to manage metric
transformation.

jpkrohling commented on receiver/mongodbatlasreceiver/receiver.go:
```diff
@@ -84,10 +86,95 @@ func (s *receiver) Shutdown(ctx context.Context) error {
+						)
+					}
+				} else {
+					s.log.Debug("Metrics not enabled")
```
Can you add this check at the top, short-circuiting this whole function? Otherwise, you'll do unnecessary processing for each process.

zenmoto committed:
address review comments

MongoDB Atlas wrapper returns errors rather than logging
change resource attributes to use conventions
modify error handling

tigrannajaryan commented:
Please resolve the conflicts.

zenmoto committed:
fix capitalization on error messages

zenmoto committed:
go mod tidy

zenmoto committed:
poll should return an error, not log

zenmoto commented:
@jpkrohling I think I addressed your concerns- is this looking right to you?

jpkrohling commented:
I'll review this again tomorrow morning, sorry for the delay.

jpkrohling reviewed (approved):
LGTM, just please confirm that the print in the stub func at the mongodb_atlas_client should still be like it is.

bogdandrutu reviewed (commented):
Please rebase since the core receiver changed to use scraperhelper.

zenmoto committed:
changes to work with scraperhelper

tigrannajaryan reviewed (approved):
LGTM.
@bogdandrutu do you have other comments on scraper or good to merge?
